api/envoy/config/route/v3/route_components.proto

```diff
@@ -1564,3 +1567,74 @@ message QueryParameterMatcher {
     bool present_match = 6;
   }
 }
+
+// HTTP Internal Redirect :ref:`architecture overview <arch_overview_internal_redirects>`.
+// [#next-free-field: 6]
+message InternalRedirectPolicy {
+  // Configures :ref:`internal redirect <arch_overview_internal_redirects>` behavior.
+  enum InternalRedirectAction {
+    // Redirect responses are proxied to downstream without triggering Internal Redirect.
+    // This is the default value.
+    PASS_THROUGH_INTERNAL_REDIRECT = 0;
+
+    // Redirect responses (specified via :ref:`redirect_response_codes
+    // <envoy_api_field_config.route.v3.InternalRedirectPolicy.redirect_response_codes>`),
+    // are followed by Envoy to the new URL. The redirect responses are not visible to
+    // the downstream unless the redirect following is rejected by other criteria.
+    HANDLE_INTERNAL_REDIRECT = 1;
+  }
+
+  // HTTP scheme combinations that are allowed to be handled as internal
+  // redirect. Downstream scheme comes first and redirect target URL scheme
+  // follows.
+  enum DownstreamAndRedirectTargetSchemePair {
+    HTTP_HTTP = 0;
+    HTTP_HTTPS = 1;
+    HTTPS_HTTP = 2;
+    HTTPS_HTTPS = 3;
+  }
+
+  message Predicate {
+    string name = 1 [(validate.rules).string = {min_bytes: 1}];
```

Comment:
We can remove name from new extensions, as they are looked up by type_url.

Reply:
Done.

Reply:
@mattklein123 the name isn't needed anymore to identify the extension. Do we still want tit around for additional metadata? We do this in filter chains as we might have multiple instanes of a filter, could we have multiple instances of a predicate here?

Reply:
Hmm ok. I've been telling people to not add this for a while, but that makes sense that we might still want the name. If that's the case should there be some common message/struct that contains both a name and typed_config with clear docs vs. duplicating everywhere?

Reply:
Yes please!